src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/migration/Cas1LostBedsToOutOfServiceBedsMigrationJob.kt

```diff
@@ -0,0 +1,118 @@
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1OutOfServiceBedCancellationRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1OutOfServiceBedEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1OutOfServiceBedReasonRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1OutOfServiceBedRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1OutOfServiceBedRevisionChangeType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1OutOfServiceBedRevisionEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1OutOfServiceBedRevisionRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.Cas1OutOfServiceBedRevisionType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.LostBedsEntity
+import java.time.OffsetDateTime
+import java.util.UUID
+
+@Suppress("detekt:LongParameterList")
+class Cas1LostBedsToOutOfServiceBedsMigrationJob(
+  private val lostBedRepository: LostBedMigrationRepository,
+  private val outOfServiceBedRepository: Cas1OutOfServiceBedRepository,
+  private val outOfServiceBedCancellationRepository: Cas1OutOfServiceBedCancellationRepository,
+  private val outOfServiceBedReasonRepository: Cas1OutOfServiceBedReasonRepository,
+  private val outOfServiceBedRevisionRepository: Cas1OutOfServiceBedRevisionRepository,
+  private val transactionTemplate: TransactionTemplate,
+  private val logger: MigrationLogger,
+) : MigrationJob() {
+  override val shouldRunInTransaction = true
+
+  override fun process() {
+    val cas1LostBeds = lostBedRepository.findAllCas1ForMigration()
+    logger.info("Starting migrating ${cas1LostBeds.count()} CAS1 lost beds")
+
+    cas1LostBeds.forEach { lostBed ->
+      transactionTemplate.executeWithoutResult {
+        createOutOfServiceBed(lostBed)
+      }
+    }
+
+    logger.info("Finished migrating ${cas1LostBeds.count()} CAS1 lost beds")
+  }
+
+  private fun createOutOfServiceBed(lostBed: LostBedsEntity) {
+    val reason = outOfServiceBedReasonRepository.findByIdOrNull(lostBed.reason.id)
+      ?: throw MigrationException(
+        "Could not find out-of-service bed reason with ID ${lostBed.reason.id}. " +
+          "Make sure that the 'update_cas1_out_of_service_bed_reasons' migration job has been run first.",
+      )
+
+    logger.info(
+      "Migrating CAS1 lost bed {id: ${lostBed.id}, premises: ${lostBed.premises.id}, bed: ${lostBed.bed.id}}",
+    )
+
+    val outOfServiceBed = outOfServiceBedRepository.saveAndFlush(
+      Cas1OutOfServiceBedEntity(
+        id = lostBed.id,
+        premises = lostBed.premises as ApprovedPremisesEntity,
+        bed = lostBed.bed,
+        createdAt = OffsetDateTime.now(),
+        cancellation = null,
+        revisionHistory = mutableListOf(),
+      ),
+    )
+
```

Comment:
do you think, that for full information in case we need to unpick any problem in the future, we ought to record the fact that the `Cas1OutOfServiceBedEntity` with the given (LostBed) id has indeed been created. e.g. with another line of logging such as:

```
"Created CAS1 out-of-service bed {id: ${outOfServiceBed.id}, premises: ${outOfServiceBed.premises.id}, bed: ${l outOfServiceBed.bed.id}}"
```

and similarly to log the successful creation of:

- `Cas1OutOfServiceBedCancellationEntity` where appropariate
- `Cas1OutOfServiceBedRevisionEntity` in every case

Reply:
I've added some additional logging to capture this.